More Documentation Updates (#280)
abcxyz/guardian: README.md

**This is not an official Google product.**

## Support

> For support related item please open a [GitHub Issue](https://github.com/abcxyz/guardian/issues/new/choose).

Reviewer: ```suggestions
For support related items, please open a [GitHub Issue](https://github.com/abcxyz/guardian/issues/new/choose).
```

- [x] Require signed commits (optional)
- [x] Require linear history

### Using Private Repositories as Modules
If you want to use Terraform modules located in private GitHub repositories then you will need to
configure git with the necessary permissions in order for Guardian to access these modules.

```shell
USERNAME=user-defined-name
TOKEN=your-token-with-read-acccess
git config --global url."https://${USERNAME}:${TOKEN}@github.com".insteadOf "https://github.com"
```

The `USERNAME` variable has no functional purpose and can be an arbitrary string. It is required

> to be present in the URL, but GitHub will identify the user based on the token.

Reviewer: ```suggestion
to be present in the URL, but GitHub will identify the user based on the token. It is recommended to provide a descriptive username and/or comment here so that developers know where this token came from (e.g. guardian-pat-token).
```

> You can use [github-token-minter](https://github.com/abcxyz/github-token-minter) to mint tokens on demand for this purpose in your GitHub Actions.

Reviewer: ```suggestion
We recommend using [github-token-minter](https://github.com/abcxyz/github-token-minter) to generate short-lived access tokens token on demand for this purpose in your GitHub Actions.
```
